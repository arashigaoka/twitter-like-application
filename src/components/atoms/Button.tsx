type ButtonProps = {
  label: string;
};

export function Button({ label }: ButtonProps): JSX.Element {
  return (
    <button
      type="button"
      className="bg-green-400 hover:bg-green-700 text-white p-2 rounded"
    >
      {label}
    </button>
  );
}
