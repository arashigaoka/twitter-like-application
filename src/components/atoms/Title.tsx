type TitleProps = {
  label: string;
};

export function Title({ label }: TitleProps): JSX.Element {
  return <h1 className="text-bold text-2xl">{label}</h1>;
}
