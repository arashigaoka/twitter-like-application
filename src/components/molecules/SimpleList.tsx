type SimpleListProps = {
  labelList: Array<string>;
};
export function SimpleList({ labelList }: SimpleListProps): JSX.Element {
  return (
    <ul className="mb-4">
      {labelList.map((label, i) => (
        // eslint-disable-next-line react/no-array-index-key
        <li key={i} className="list-disc">
          {label}
        </li>
      ))}
    </ul>
  );
}
