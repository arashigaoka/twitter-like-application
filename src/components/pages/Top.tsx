import Head from 'next/head';
import { Button } from '../atoms/Button';
import { Card } from '../organisms/Card';
import { useTop } from './use-top';

export const Top = (): JSX.Element => {
  const { posts, addPosts, content, setContent } = useTop();
  return (
    <>
      <Head>
        <link rel="icon" href="/favicon.ico" />
      </Head>
      <div className="conntainer p-20">
        <h1 className="text-2xl bold">My Twitter Like Application</h1>
        <div className="flex space-x-10">
          <div className="mt-10">
            <h2 className="text-xl text-gray-700">Add Post</h2>
            <textarea
              cols={30}
              rows={3}
              className="block rounded"
              value={content}
              onChange={(e) => setContent(e.target.value)}
              placeholder="Let's add posts!"
            />
            <Button label="Submit" clickHandler={addPosts} />
          </div>
          <div className="mt-10">
            <h2 className="text-xl text-gray-700">All Posts</h2>
            <ul className="space-y-2">
              {posts.map((post) => (
                <li key={post.id}>
                  <Card post={post} />
                </li>
              ))}
            </ul>
          </div>
        </div>
      </div>
    </>
  );
};
