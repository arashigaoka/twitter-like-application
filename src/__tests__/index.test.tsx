import TestRenderer from 'react-test-renderer';
import { Button } from '../components/atoms/Button';
import { SimpleList } from '../components/molecules/SimpleList';
import Top from '../pages/index';

describe('test', () => {
  test('renders properly', () => {
    const testRenderer = TestRenderer.create(<Top />);
    const button = testRenderer.root.findByType(Button);
    expect(button.props.label).toBe('Click Me!');
    const simpleList = testRenderer.root.findByType(SimpleList);
    expect(simpleList.props.labelList.length).toBe(4);
  });
});
