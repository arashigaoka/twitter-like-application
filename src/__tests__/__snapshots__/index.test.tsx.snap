// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test renders properly 1`] = `
<div>
  loading...
</div>
`;
